nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/main/java/org/apache/nifi/lookup/RestLookupService.java

```diff
@@ -0,0 +1,467 @@
+import org.apache.nifi.attribute.expression.language.Query;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.components.Validator;
+import org.apache.nifi.controller.AbstractControllerService;
+import org.apache.nifi.controller.ConfigurationContext;
+import org.apache.nifi.expression.ExpressionLanguageScope;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.proxy.ProxyConfiguration;
+import org.apache.nifi.proxy.ProxyConfigurationService;
+import org.apache.nifi.proxy.ProxySpec;
+import org.apache.nifi.record.path.FieldValue;
+import org.apache.nifi.record.path.RecordPath;
+import org.apache.nifi.record.path.validation.RecordPathValidator;
+import org.apache.nifi.schema.access.SchemaNotFoundException;
+import org.apache.nifi.serialization.MalformedRecordException;
+import org.apache.nifi.serialization.RecordReader;
+import org.apache.nifi.serialization.RecordReaderFactory;
+import org.apache.nifi.serialization.SimpleRecordSchema;
+import org.apache.nifi.serialization.record.MapRecord;
+import org.apache.nifi.serialization.record.Record;
+import org.apache.nifi.serialization.record.RecordSchema;
+import org.apache.nifi.ssl.SSLContextService;
+import org.apache.nifi.util.StringUtils;
+
+import javax.net.ssl.SSLContext;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.Proxy;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static org.apache.commons.lang3.StringUtils.trimToEmpty;
+
+@Tags({ "rest", "lookup", "json", "xml", "http" })
+@CapabilityDescription("Use a REST service to enrich records.")
+@DynamicProperties({
+    @DynamicProperty(name = "*", value = "*", description = "All dynamic properties are added as HTTP headers with the name " +
+            "as the header name and the value as the header value.")
+})
+public class RestLookupService extends AbstractControllerService implements RecordLookupService {
+    static final PropertyDescriptor BASE_URL = new PropertyDescriptor.Builder()
+        .name("rest-lookup-base-url")
+        .displayName("Base URL")
+        .description("The base URL for the REST endpoint. Expression language is evaluated against variable registry." +
+                " This property can be used to resolve environment specific part of the URL." +
+                " The result string is prepended to the 'URL'." +
+                " See 'Additional Details' to see an example.")
+        .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+        .required(false)
+        .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+        .build();
+
+    static final PropertyDescriptor URL = new PropertyDescriptor.Builder()
```

Comment:
I don't understand why we have two different properties here for "URL" and "Base URL". Can you elaborate? If it's because we want to evaluate the Expression Language differently, then we can still do that by just using `context.getProperty(URL).evaluateExpressionLanguage(coordinates);` and that will evaluate against both the coordinates the Variable Registry...

Reply:
@markap14 Thanks, that is correct! I added "Base URL" because I thought it is not possible to use Variable Registry, but that was because it uses PreparedQuery instead of PropertyValue. By using PropertyValue, it can use Variable Registry, too.

@MikeThomsen I will send a PR again to fix this.

Reply:
@MikeThomsen Sent a PR to your branch to fix this. Please merge.
https://github.com/MikeThomsen/nifi/pull/2/commits